cpp/src/aztec/dsl/acir_format/ecdsa_secp256k1.cpp

```diff
@@ -0,0 +1,122 @@
+    secp256k1_ct::fq_ct y(witness_ct(ctx, barretenberg::fr(y_u256.slice(0, secp256k1_ct::fq_ct::NUM_LIMB_BITS * 2))),
+                          witness_ct(ctx,
+                                     barretenberg::fr(y_u256.slice(secp256k1_ct::fq_ct::NUM_LIMB_BITS * 2,
+                                                                   secp256k1_ct::fq_ct::NUM_LIMB_BITS * 4))));
+
+    return { x, y };
+}
+
+// vector of bytes here, assumes that the witness indices point to a field element which can be represented
+// with just a byte.
+// notice that this function truncates each field_element to a byte
+byte_array_ct ecdsa_vector_of_bytes_to_byte_array(plonk::TurboComposer& composer, std::vector<uint32_t> vector_of_bytes)
+{
+    byte_array_ct arr(&composer);
+
+    // Get the witness assignment for each witness index
+    // Write the witness assignment to the byte_array
+    for (const auto& witness_index : vector_of_bytes) {
+
+        field_ct element = field_ct::from_witness_index(&composer, witness_index);
+        size_t num_bytes = 1;
+
+        byte_array_ct element_bytes(element, num_bytes);
+        arr.write(element_bytes);
+    }
+    return arr;
+}
+witness_ct ecdsa_index_to_witness(plonk::TurboComposer& composer, uint32_t index)
+{
+    fr value = composer.get_variable(index);
+    return { &composer, value };
+}
+
+void create_ecdsa_verify_constraints(plonk::TurboComposer& composer, const EcdsaSecp256k1Constraint& input)
+{
+
+    auto new_sig = ecdsa_convert_signature(composer, input.signature);
+
+    auto message = ecdsa_vector_of_bytes_to_byte_array(composer, input.message);
+    auto pub_key_x_byte_arr = ecdsa_vector_of_bytes_to_byte_array(composer, input.pub_x_indices);
+    auto pub_key_y_byte_arr = ecdsa_vector_of_bytes_to_byte_array(composer, input.pub_y_indices);
+
+    auto pub_key_x_fq = secp256k1_ct::fq_ct(pub_key_x_byte_arr);
+    auto pub_key_y_fq = secp256k1_ct::fq_ct(pub_key_y_byte_arr);
+
+    std::vector<uint8_t> rr(new_sig.r.begin(), new_sig.r.end());
+    std::vector<uint8_t> ss(new_sig.s.begin(), new_sig.s.end());
+
+    stdlib::ecdsa::signature<plonk::TurboComposer> sig{ stdlib::byte_array<plonk::TurboComposer>(&composer, rr),
+                                                        stdlib::byte_array<plonk::TurboComposer>(&composer, ss) };
+
+    pub_key_x_fq.assert_is_in_field();
+    pub_key_y_fq.assert_is_in_field();
+
+    // TODO: crypto-dev to fix calculation and constraining of the signature result is correct
+    // the above line is currently a placeholder as unused variabels are not allowed in the build
+    // auto pub_key = secp256k1_ct::g1_ct(pub_key_x_fq, pub_key_y_fq);
+    // bool_ct signature_result = stdlib::ecdsa::
+    //     verify_signature<plonk::TurboComposer, secp256k1_ct::fq_ct, secp256k1_ct::bigfr_ct,
+    //     secp256k1_ct::g1_bigfr_ct>(
+    //         message, pub_key, sig);
+
+    // auto result_bool = composer.add_variable(signature_result.get_value() == true);
```

Comment:
cc @kevaundray @vezenovm As discussed in #204 - this needs to be fixed here so this can land and be correct. Not sure who to ping for help.

Reply:
I pushed a change to use the correct interface

Reply:
Thanks @kevaundray looks like I was using the wrong g1 interface when constructing the pub key